python/ray/train/v2/_internal/state/state_manager.py

```diff
@@ -0,0 +1,237 @@
+    ) -> None:
+        def _convert_worker(worker: Worker) -> TrainWorker:
+
+            actor: ActorHandle = worker.actor
+            distributed_context: DistributedContext = worker.distributed_context
+            actor_metadata: ActorMetadata = worker.metadata
+
+            return TrainWorker(
+                world_rank=distributed_context.world_rank,
+                local_rank=distributed_context.local_rank,
+                node_rank=distributed_context.node_rank,
+                actor_id=actor._actor_id.hex(),
+                node_id=actor_metadata.node_id,
+                node_ip=actor_metadata.node_ip,
+                pid=actor_metadata.pid,
+                gpu_ids=actor_metadata.gpu_ids,
+                status=ActorStatus.ALIVE,
+                resources=TrainResources(resources=worker.resources),
+            )
+
+        workers: List[TrainWorker] = [_convert_worker(worker) for worker in workers]
+
+        run_attempt = self._run_attempts[run_id][attempt_id]
+        run_attempt.status = RunAttemptStatus.RUNNING
+        run_attempt.status_detail = None
+        run_attempt.workers = workers
+        self._create_or_update_train_run_attempt(run_attempt)
+
+    def update_train_run_attempt_finished(
+        self,
+        run_id: str,
+        attempt_id: str,
+    ):
+        run_attempt = self._run_attempts[run_id][attempt_id]
+        run_attempt.status = RunAttemptStatus.FINISHED
+        run_attempt.status_detail = None
+        run_attempt.end_time_ms = _current_time_ms()
+        self._create_or_update_train_run_attempt(run_attempt)
+
+    def update_train_run_attempt_errored(
+        self,
+        run_id: str,
+        attempt_id: str,
+        status_detail: str,
+    ):
+        run_attempt = self._run_attempts[run_id][attempt_id]
+        run_attempt.status = RunAttemptStatus.ERRORED
+        run_attempt.status_detail = status_detail
+        run_attempt.end_time_ms = _current_time_ms()
+        self._create_or_update_train_run_attempt(run_attempt)
+
+    def update_train_run_attempt_aborted(
+        self,
+        run_id: str,
+        attempt_id: str,
+    ):
+        run_attempt = self._run_attempts[run_id][attempt_id]
+        run_attempt.status_detail = None  # TODO: Add status detail.
+        run_attempt.status = RunAttemptStatus.ABORTED
+        run_attempt.end_time_ms = _current_time_ms()
+        self._create_or_update_train_run_attempt(run_attempt)
+
+    # TODO: Add export API functionality to these. Or maybe in Actor.
+
+    def _create_or_update_train_run(self, run: TrainRun) -> None:
+        object_ref = self._state_actor.create_or_update_train_run.remote(run)
+        self._safe_get(object_ref)
+
+    def _create_or_update_train_run_attempt(self, run_attempt: TrainRunAttempt) -> None:
+        object_ref = self._state_actor.create_or_update_train_run_attempt.remote(
+            run_attempt
+        )
+        self._safe_get(object_ref)
```

Comment:
Actually, I think these should be a fire and forget so we don't ever block the train controller on `ray.get` calls.